src/consumer.ts

```diff
@@ -142,7 +143,36 @@ export class Consumer extends TypedEventEmitter {
       this.emit('aborted');
     }
 
-    this.emit('stopped');
+    if (options?.waitForInFlightMessagesMs > 0) {
+      this.waitForInFlightMessagesToComplete(
+        options.waitForInFlightMessagesMs
+      ).then(() => {
+        this.emit('stopped');
+      });
+    } else {
+      this.emit('stopped');
+    }
+  }
+
+  /**
+   * Wait for in flight messages to complete.
+   * @param {number} waitTimeoutMs
+   * @private
+   */
+  private async waitForInFlightMessagesToComplete(
+    waitTimeoutMs: number
+  ): Promise<void> {
+    const startedAt = Date.now();
+    while (this.isPolling) {
+      if (Date.now() - startedAt > waitTimeoutMs) {
+        logger.debug(
```

Comment:
Couldn't this just be in a timeout and then we wouldn't need the date stuff?

Reply:
I wasn't sure how to address this feedback request.  We need some way to track that we haven't waited too long.  I've refactored the code to use `setTimeout` rather than a `Promise` loop, but I'm open to suggestion on how to avoid using dates for the timeout exceeded calculation.